op-batcher/batcher/channel_manager.go

```diff
@@ -120,23 +115,23 @@ func (s *channelManager) TxConfirmed(_id txID, inclusionBlock eth.BlockID) {
 	s.log.Debug("marked transaction as confirmed", "id", id, "block", inclusionBlock)
 }
 
-// removePendingChannel removes the given completed channel from the manager's state.
-func (s *channelManager) removePendingChannel(channel *channel) {
-	if s.currentChannel == channel {
-		s.currentChannel = nil
-	}
-	index := -1
-	for i, c := range s.channelQueue {
-		if c == channel {
-			index = i
+func (s *channelManager) rewindToBlockWithHash(blockHash common.Hash) {
+	for i, b := range s.blocks {
+		if b.Hash() == blockHash {
+			s.blockCursor = i
```

Comment:
similar to frames, do we need to guard against the block cursor moving forward?

Reply:
Good catch. Yes I think so, if for no other reason than this being true to the notion of "rewinding".